test-data/unit/check-tuples.test

```diff
@@ -1241,3 +1241,43 @@ b = ("bar", 7)
 reveal_type(a + b)  # N: Revealed type is 'Tuple[builtins.int, builtins.str, builtins.int, builtins.str, builtins.int]'
 
 [builtins fixtures/tuple.pyi]
+
+[case testAssigningWithLongTupleInitializer]
+from typing import Tuple
+
+# long initializer assignment with few mismatches
+t: Tuple[int, ...] = (
+    1, 2, 3, 4, 5, 6, 7, 8, "str", "str", "str", 11, 12, 14, 15,  # E: Incompatible types in assignment (expression Tuple item 8 has type "str"; "int" expected; expression Tuple item 9 has type "str"; "int" expected; expression Tuple item 10 has type "str"; "int" expected; Total 3 items are incompatible; 0 items are omitted)
```

Comment:
I think that this is too much information in a single error message. It makes the error message hard to read. We generally split long messages into an error followed by one or more notes. In this case, something like could make sense (it doesn't have to be exactly like this):

```
...: error: Incompatible types in assignment (3 tuple items are incompatible)
...: note: Expression tuple item 8 has type "str"; "int" expected
...: note: Expression tuple item 9 has type "str"; "int" expected
...: note: Expression tuple item 10 has type "str"; "int" expected
```

In a test case you can write something like this to represent this:
```
<code> \
    # E: <message> \
    # N: <note>
...
```

The part "0 items are omitted" seems redundant. It can be left out if no items are omitted.

Even better, the notes could use the line/columns with the tuple item expression as context (when possible; clearly this is not always feasible). This would be particularly helpful when the tuple initializer spans several lines.

Reply:
I tried to directly use tuple item from e.g. `subtype.items` as context and no row/column info is shown. I guess I may use it in an incorrect way.